From 7def05516a2f4971d9f388df406d9fc2774d06e2 Mon Sep 17 00:00:00 2001
From: Soumya Aithal <soumya.aithal@lge.com>
Date: Tue, 12 Sep 2017 11:41:27 -0700
Subject: [PATCH] Fix avoutputd, lsm start sequence issue

:Release Notes:
Fixed issue regarding unable to launch webapps if avoutputd started
before surface manager.

:Detailed Notes:
This fix calls drmSetMaster to acquire drm master status so it can
authenticate legacy wayland clients which rely on magic token exchange
between client and server for gpu access.

:Testing Performed:
Tested that apps were launched after changing the startup sequence.

:QA Notes:

:Issues Addressed:
[PLAT-45032] Investigate app launching issue when avoutputd is
    started before surface manager

[GF-9999] Summary
---
 .../eglfs/deviceintegration/eglfs_kms/qeglfskmsdevice.cpp         | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsdevice.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsdevice.cpp
index b7cb814..d87a4e0 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsdevice.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsdevice.cpp
@@ -1,3 +1,4 @@
+
 /****************************************************************************
 **
 ** Copyright (C) 2015 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
@@ -389,7 +390,12 @@ bool QEglFSKmsDevice::open()
         return false;
     }
 
-    return true;
+
+    if (drmSetMaster(m_dri_fd))
+    {
+        qWarning("Failed to set drmSetMaster");
+    }
+   return true;
 }
 
 void QEglFSKmsDevice::close()
